upup/pkg/fi/cloudup/loader.go

```diff
@@ -214,6 +217,40 @@ func (l *Loader) addAssetCopyTasks(assets []*assets.ContainerAsset, lifecycle *f
 
 		}
 	}
+
+	return nil
+}
+
+// addAssetFileCopyTasks creates the new tasks for copying files.
+func (l *Loader) addAssetFileCopyTasks(assets []*assets.FileAsset, lifecycle *fi.Lifecycle) error {
+	for _, asset := range assets {
+
+		// test if the asset needs to be copied
+		if asset.CanonicalFileURL != nil && asset.FileURL != asset.CanonicalFileURL {
+			glog.V(10).Infof("processing asset: %q, %q", asset.FileURL.String(), asset.CanonicalFileURL.String())
+			context := &fi.ModelBuilderContext{
+				Tasks: l.tasks,
+			}
+
+			glog.V(10).Infof("adding task: %q", asset.FileURL.String())
+
+			copyFileTask := &assettasks.CopyFile{
+				Name:       fi.String(asset.CanonicalFileURL.String()),
+				TargetFile: fi.String(asset.FileURL.String()),
+				SourceFile: fi.String(asset.CanonicalFileURL.String()),
+				SourceSHA:  fi.String(asset.SHAValue),
+				Lifecycle:  lifecycle,
+			}
+
+			// TODO: we are getting a duplicate task here, not certain why, we should track it down
```

Comment:
Are we getting a duplicate task still?  Which one?  Is there a test which shows it?

Reply:
I think you found it ... will test